packages/components/src/components/Address/Address.tsx

```diff
@@ -6,11 +6,27 @@ export interface AddressProps {
    * The address to display
    */
   value: string;
+  /**
+   * boolean to shorten the address if addresses does not resolve to an ENS name
+   */
+  shortened: boolean;
 }
 
 /**
  * A component to display an address
  */
-export const Address: React.FC<AddressProps> = ({ value }) => {
-  return <Input value={value} />;
+export const Address: React.FC<AddressProps> = ({ value, shortened }) => {
+  if (shortened) {
+    if (value.includes('.eth')) {
+      return <Input value={value} />;
+    } else if (value === '' || value === 'Not connected') {
+      return <Input value={value} />;
+    } else {
+      let address: string;
+      address = `${value.substring(0, 4)}...${value.substring(value.length - 4)}`;
+      return <Input value={address.toLowerCase()} />;
+    }
+  } else {
+    return <Input value={value} />;
```

Comment:
on a fundamental level, it may be worth having a single `return <Input value={%_var_%} />;` at the end of this component. And focus the conditional logic to assign a respective value to a `let` variable

e.g.

```js
let displayAddress;

if(case) {
 displayAddress = 'a';
} else if (case2) {
 displayAddress = 'b';
}

return <Input value={displayAddress} />